programs/drift/src/instructions/lp_pool.rs

```diff
@@ -0,0 +1,1879 @@
+
+pub fn handle_view_lp_pool_swap_fees<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, ViewLPPoolSwapFees<'info>>,
+    in_market_index: u16,
+    out_market_index: u16,
+    in_amount: u64,
+    in_target_weight: i64,
+    out_target_weight: i64,
+) -> Result<()> {
+    let slot = Clock::get()?.slot;
+    let state = &ctx.accounts.state;
+
+    let lp_pool = &ctx.accounts.lp_pool.load()?;
+    let in_constituent = ctx.accounts.in_constituent.load()?;
+    let out_constituent = ctx.accounts.out_constituent.load()?;
+    let constituent_correlations: AccountZeroCopy<'_, i64, ConstituentCorrelationsFixed> =
+        ctx.accounts.constituent_correlations.load_zc()?;
+
+    let AccountMaps {
+        perp_market_map: _,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        &mut ctx.remaining_accounts.iter().peekable(),
+        &MarketSet::new(),
+        &MarketSet::new(),
+        slot,
+        Some(state.oracle_guard_rails),
+    )?;
+
+    let in_spot_market = spot_market_map.get_ref(&in_market_index)?;
+    let out_spot_market = spot_market_map.get_ref(&out_market_index)?;
+
+    let in_oracle_id = in_spot_market.oracle_id();
+    let out_oracle_id = out_spot_market.oracle_id();
+
+    let (in_oracle, _) = oracle_map.get_price_data_and_validity(
+        MarketType::Spot,
+        in_spot_market.market_index,
+        &in_oracle_id,
+        in_spot_market.historical_oracle_data.last_oracle_price_twap,
+        in_spot_market.get_max_confidence_interval_multiplier()?,
+        0,
+    )?;
+    let in_oracle = in_oracle.clone();
+
+    let (out_oracle, _) = oracle_map.get_price_data_and_validity(
+        MarketType::Spot,
+        out_spot_market.market_index,
+        &out_oracle_id,
+        out_spot_market
+            .historical_oracle_data
+            .last_oracle_price_twap,
+        out_spot_market.get_max_confidence_interval_multiplier()?,
+        0,
+    )?;
+
+    let (in_amount, out_amount, in_fee, out_fee) = lp_pool.get_swap_amount(
+        &in_oracle,
+        &out_oracle,
+        &in_constituent,
+        &out_constituent,
+        &in_spot_market,
+        &out_spot_market,
+        in_target_weight,
+        out_target_weight,
+        in_amount as u128,
+        constituent_correlations.get_correlation(
+            in_constituent.constituent_index,
+            out_constituent.constituent_index,
+        )?,
+    )?;
+    msg!(
+        "in_amount: {}, out_amount: {}, in_fee: {}, out_fee: {}",
+        in_amount,
+        out_amount,
+        in_fee,
+        out_fee
+    );
+    Ok(())
+}
+
+#[access_control(
+    fill_not_paused(&ctx.accounts.state)
+)]
+pub fn handle_lp_pool_add_liquidity<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, LPPoolAddLiquidity<'info>>,
+    in_market_index: u16,
+    in_amount: u128,
+    min_mint_amount: u64,
+) -> Result<()> {
+    #[cfg(not(feature = "anchor-test"))]
+    validate!(
+        DLP_WHITELIST.contains(&ctx.accounts.authority.key()),
```

Comment:
would use the same logic we used to whitelist drift access via a mint so we dont need to use prog deploys